### PR TITLE
CMS-437: Auto merge old PRs.

### PR DESCRIPTION
**Pull request by kporras07**
### Overview
This pull request adds a command to auto merge old PRs.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes 
| Has tests?    | no
| BC breaks?    | no     
| Deprecations? | no 

Depends on https://github.com/g1a/hubph/pull/47 and then update hubph in this tool

### CODE REVIEW

File: src/Cli/OrgCommands.php [hunk 1]
```diff
@@ -251,6 +255,71 @@ public function orgUpdateProjectsInfo($csv_file, $options = [
...
+        'pr-title-query' => '[UpdateTool - Project Information]',
+        'pr-title-pattern' => '',
+    ])
+    {
+        $api = $this->api($options['as']);
+        $prTitleQuery = $options['pr-title-query'];
+        $prTitlePattern = $options['pr-title-pattern'];
+        $age = $options['age'];
+
+        if (file_exists($csv_file)) {
```
Comment (greg-1-anderson) [inline src/Cli/OrgCommands.php:274]: Can we use the GitHub API to find the open PRs? Can we search for PRs across all of the projects in an organization,  and match by query, or do we have to search for PRs in a specific project?

File: src/Cli/OrgCommands.php [hunk 2]
```diff
@@ -251,6 +255,71 @@ public function orgUpdateProjectsInfo($csv_file, $options = [
...
+                    continue;
+                }
+                if (empty($row[0])) {
+                    // Empty line, probably EOF. Break loop.
+                    break;
+                }
+                $projectName = $row[2];
+                $projectFullName = $row[3];
+                $projectOrg = $row[5];
+                $projectSupportLevel = $row[23];
```
Comment (greg-1-anderson) [inline src/Cli/OrgCommands.php:289]: If we can't search across the whole org & find PRs in any project, then we should find the column numbers by label, as is done in the other csv file command.

File: src/Cli/OrgCommands.php (hunk 1 above)
Comment (greg-1-anderson) [reply src/Cli/OrgCommands.php:274]: It looks like this is available, e.g. https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Apantheon-systems+archived%3Afalse+

Did not find API equivalent, but should be doable. Might need a new hubph wrapper function, but I'd guess it's possible with the knplabs github api.

Comment (greg-1-anderson) [reply src/Cli/OrgCommands.php:274]: Then we can just forget about the cvs file in this command.

**Comment by greg-1-anderson** [comment]
This is looking good.

- A "dry run" mode would be useful
- When implementing the ability to skip PRs for badges that are actively maintained, consider ways that we could conveniently do this that would be generic, so that we could potentially re-use this command for different similar uses in the future. For example, it could take an option with a pattern that must or must not appear in the PR diff in order for the PR to be eligible for merging.

**Comment by kporras07** [comment]
@greg-1-anderson I added 3 new options:
- --diff-include-pattern
- --diff-exclude-pattern
- --dry-run

First 2 options accept a regex pattern that will search in the diff and exit if it doesn't comply with what is expected/not expected. Dry-run does everything but the actual merge.

**Review by greg-1-anderson — APPROVED** [review]
Awesome

**Pull request merged**